src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -275,8 +275,8 @@ private void fireAuditFinished() {
      */
     private void processFiles(List<File> files) throws CheckstyleException {
         for (final File file : files) {
+            final String fileName = file.getAbsolutePath();
```

Comment:
This can't throw an exception right?

Reply:
throws SecurityException as declared - https://docs.oracle.com/javase/8/docs/api/java/io/File.html#getAbsolutePath--

but can probably throw any other exception if smth goes completely wrong.

Reply:
Then I will have to add another try/catch as if any of these exceptions occur file being processed will be hidden.

Reply:
I looked at the source for `file.getAbsolutePath` and the security exception does throw the path in it.
````
                    /* Don't disclose the drive's directory in the exception */
                    throw new SecurityException("Cannot resolve path " + path);
````
and even the last resort has:
````
throw new InternalError("Unresolvable path: " + path);
````

I moved the method call back in the try/catch and added support for `null` file names.